### PR TITLE
Warn if app-backend can't start-up because missing dir

### LINKED ISSUE #3412: Backend failed to start up [Error: ENOENT: no such file or directory, scandir '/Users/name/dev/external/backstage/packages/app/dist/static'
## Expected Behavior

I can start the `packages/backend` of this repository when I make a clean checkout of this repository or after running `yarn clean`.

## Current Behavior

While working on Backstage (this repository, not my own instance!) I often run into the following error when doing `yarn dev` or `yarn start-backend`:

```
2020-11-24T09:06:28.506Z app info Serving static app content from /Users/name/dev/external/backstage/packages/app/dist
[1] Backend failed to start up [Error: ENOENT: no such file or directory, scandir '/Users/name/dev/external/backstage/packages/app/dist/static'] {
[1]   errno: -2,
[1]   code: 'ENOENT',
[1]   syscall: 'scandir',
[1]   path: '/Users/name/dev/external/backstage/packages/app/dist/static'
[1] }
```

The problem is, that I haven't compiled the frontend (and I don't need to locally as I'm using `yarn dev`) and therefore the folder `/dist/static` is missing in `packages/app`. The `plugin-app-backend` expects that folder to be their to serve its content.

## Possible Solution

There are several solutions:

* Manually create empty `/dist/static` after running `yarn clean`.
* Making sure that `/dist/static` exists in the `yarn dev` or `yarn start-backend` command
* Removing the `plugin-app-backend` plugin, but it's good that we show the usage here.
* Changing `plugin-app-backend` to allow the missing folder. This is in my option a bad choice as it will hide real errors if the compiled code is really missing.

### PR DESCRIPTION
**Pull request by Fox32**
Fixes #3412 

I tried to add a test for it (it already has a nice one!) but I can't reproduce it there...

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)

### CODE REVIEW

File: plugins/app-backend/src/service/router.ts [hunk 1]
```diff
@@ -21,6 +21,7 @@ import { Logger } from 'winston';
 import { notFoundHandler, resolvePackagePath } from '@backstage/backend-common';
 import { Config } from '@backstage/config';
 import { injectConfig, readConfigs } from '../lib/config';
+import fs from 'fs';
```
Comment (Rugvip) [inline plugins/app-backend/src/service/router.ts:24]: Use `pathExists` from `fs-extra`!

**Review by Rugvip — COMMENTED** [review]
Thanks! Minor nit

File: plugins/app-backend/src/service/router.ts (hunk 1 above)
Comment (Fox32) [reply plugins/app-backend/src/service/router.ts:24]: Isn't that just a wrapper around `access`? https://github.com/jprichardson/node-fs-extra/blob/5b29ae3aa5198c33820a07eaa4493a5b6d01fa82/lib/path-exists/index.js#L5-L7

Comment (Fox32) [reply plugins/app-backend/src/service/router.ts:24]: But it looks nicer 😉

Comment (Rugvip) [reply plugins/app-backend/src/service/router.ts:24]: May be functionally the same, but we generally use `fs-extra` instead of `fs.promises`

Comment (Fox32) [reply plugins/app-backend/src/service/router.ts]: Done, fs-extra was available in the plugin anyway 👍

**Review by Rugvip — APPROVED** [review]
:shipit:

**Pull request merged**